lib/api_me/sorting.rb

```diff
@@ -26,20 +31,22 @@ def sort_meta
 
     protected
 
-    def sort
-      unless sort_criteria === ""
-        sort_p = sort_criteria
+    def sort(criteria = default_sort_criteria)
+      unless sort_association == ""
+        criteria_class = criteria.camelize.constantize
+
         if sort_reverse === "true"
-          self.scope = scope.sort_by {|scope| scope[sort_p]}.reverse!
+          scope.joins(criteria.to_sym).merge(criteria_class.order(sort_association => :desc))
         else
-          self.scope = scope.sort_by {|scope| scope[sort_p]}
+          scope.joins(criteria.to_sym).merge(criteria_class.order(sort_association => :asc))
```

Comment:
In fact I'd argue here you always want `left_outer_joins` and not `joins`. Also, why not split on the `.` instead of passing a joins association through the API? Just seems messy and harder to use this way. Would prefer that something where `sort_params[:criteria] = "posts.title` does something like:
```SQL
SELECT users.*, posts.title FROM users LEFT JOIN posts ON posts.author_id = users.id ...
```

You also need to select the join field which you are doing an `ORDER BY` on otherwise it wont work, like in the SQL example above.